spec/client_spec.js

```
@@ -0,0 +1,27 @@
+describe("Client", function() {
+
+    var GiantSwarm = require('../lib/client');
+
+    beforeEach(function() {
+      GiantSwarm.setApiEndpoint('https://api.giantswarm.io');
+      GiantSwarm.setAuthToken(null);
+    });
+
+    it("should allow me to ping the server", function(done){
+      GiantSwarm.ping(function(){
+        var value = 1;
+        expect(value).toEqual(1);
+        done();
+      });
+    });
+
+    it("should not allow me to ping the server", function(done){
+      GiantSwarm.setApiEndpoint('https://www.google.com');
+      GiantSwarm.ping(function(){}, function(err){
```

Same here as above. The success callback should raise an error.